sys/include/net/nanocoap.h

```diff
@@ -987,6 +990,24 @@ ssize_t coap_opt_add_opaque(coap_pkt_t *pkt, uint16_t optnum, const uint8_t *val
  */
 ssize_t coap_opt_add_uquery(coap_pkt_t *pkt, const char *key, const char *val);
 
+/**
+ * @brief   Adds a single Uri-Query option in the form 'key=value' into pkt
+ *
+ *
+ * @param[in,out] pkt         The package that is being build
+ * @param[in]     key         Key to add to the query string
+ * @param[in]     key_len     Length of @p key
+ * @param[in]     val         Value to assign to @p key (may be NULL)
+ * @param[in]     val_len     Length of @p val. 0 if @p val is NULL
+ *
+ * @pre     ((pkt != NULL) && (key != NULL))
```

Comment:
Also add `key_len > 0` since that assert is in the code.

Reply:
added `key_len > 0`